addons/main/functions/fnc_doShareInformation.sqf

```diff
@@ -32,6 +32,14 @@ if (isNull _target) then {
     _target = _unit findNearestEnemy _unit;
 };
 
+// custom handlers
+private _handlersReturn = true;
+{
+    private _handlerResult = [_unit, _target, _range, _override] call _x;
+    if (!isNil _handlerResult && _handlerResult isEqualTo false) exitWith {_handlersReturn = false};
+} forEach GVAR(shareHandlers);
+if (!_handlersReturn) exitWith {false};
```

Comment:
May I suggest negating this? The idea can be borrowed from the vanilla event handlers which allow to overwrite the handles status. Meaning returning `true` means it was handled and no further need to do something and `false` means "not handled, continue". Of course the header of the `addShareInformationHandler` functions needs to change a little.

Another change suggestion is to wrap the second condition in a code block so it actually does lazy eval.

```suggestion
private _stopShare = false;
{
    private _callbackResult = [_unit, _target, _range, _override] call _x;
    if (!(isNil "_callbackResult") && {_callbackResult isEqualTo true}) exitWith {_stopShare = true};
} forEach GVAR(shareHandlers);
if (_stopShare) exitWith {false};
```